### PR TITLE
[Example] Update SEAL+NGNN for ogbl-citation2

### PR DESCRIPTION
**Pull request by Ereboas**
## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the best of my knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change

## Changes

- [x] Add comments, and slightly adjust the structure of `SEALDataset.__init__` method, as discussed in #4772

### CODE REVIEW

File: examples/pytorch/ogb/ngnn_seal/main.py [hunk 1]
```diff
@@ -48,17 +48,16 @@ def __init__(
...
-        if not self.dynamic:
-            self.g_list, tensor_dict = self.load_cached()
-            self.labels = tensor_dict["y"]
-            return
-
         pos_edge, neg_edge = get_pos_neg_edges(
-            split, self.split_edge, self.graph, self.percent
+            self.split, self.split_edge, self.graph, self.percent
         )
         self.links = torch.cat([pos_edge, neg_edge], 0)  # [Np + Nn, 2]
```
Comment (frozenbugs) [inline examples/pytorch/ogb/ngnn_seal/main.py:54]: Semantically, the logic of the code has changed, I am wondering why it was OK to return at L54? self.links was not set in the previous code. Was it a bug?

File: examples/pytorch/ogb/ngnn_seal/main.py [hunk 2]
```diff
@@ -508,6 +501,12 @@ def print_log(*x, sep="\n", end="\n", mode="a"):
 
     if 0 < args.sortpool_k <= 1:  # Transform percentile to number.
         if args.dataset.startswith("ogbl-citation"):
+        # For this dataset, subgraphs extracted around positive edges are
```
Comment (frozenbugs) [inline examples/pytorch/ogb/ngnn_seal/main.py:504]: indent by 4 more, the code which explains the logic below should align with the code.

File: examples/pytorch/ogb/ngnn_seal/main.py (hunk 1 above)
Comment (Ereboas) [reply examples/pytorch/ogb/ngnn_seal/main.py:54]: not a bug. In the previous code, self.links would be set in method self.load_cached() at L52, and I remove the relevant code in that method (the same as L51-L54 of the new version).

**Review by frozenbugs — APPROVED** [review]

**Comment by frozenbugs** [comment]
@dgl-bot

**Pull request merged**